### PR TITLE
Idle Screen initial implementation

### PR DESCRIPTION
**Pull request by NeonDaniel**
Implements an idle screen in core, rather than an enclosure skill like in Mycroft (https://github.com/MycroftAI/skill-mark-2/blob/f873ac4f7c77662f0a23e24e0c8a70ff6d91e94b/__init__.py#L49).

The code in core should only handle displaying/updating the idle screen; any controls or device-specific components are not included here and should be added to separate skills or enclosures as appropriate.

### CODE REVIEW

File: neon_core/gui/resting_screen.py [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+                time.sleep(0.5)
+                self.schedule_event(
+                    self.show, int(offset), name="IdleCheck"
+                )
+                LOG.info("Showing idle screen in " "{} seconds".format(offset))
+            except Exception as e:
+                LOG.exception(repr(e))
+
+
+def init_resting_screen_manager():
```
Comment (JarbasAl) [inline neon_core/gui/resting_screen.py:225]: could this be integrated instead into the GUIManager class? i think that would feel cleaner

Comment (NeonDaniel) [reply neon_core/gui/resting_screen.py:225]: Good catch; missed that cleanup after porting from the skill

File: neon_core/gui/resting_screen.py [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+
+
+class RestingScreen:
+    """
+    Implementation of resting screens.
+    This class handles registration and override of resting screens,
+    encapsulating the system.
+    """
+
+    def __init__(self):
```
Comment (JarbasAl) [inline neon_core/gui/resting_screen.py:47]: can we add an optional param bus=None ? this way we can reuse the connection from GuiManager

File: neon_core/gui/resting_screen.py [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+        skill = MycroftSkill()
```
Comment (JarbasAl) [inline neon_core/gui/resting_screen.py:48]: lets also set our own skill_id after `skill.bind`, that's relevant metadata and will show up in the message.context

`skill.skill_id = "resting_screen.neon"`

File: neon_core/gui/resting_screen.py [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+        bus = MessageBusClient()
+        bus.run_in_thread()
+        skill.bind(bus)
+        self.bus = skill.bus
+        self.gui = skill.gui
```
Comment (JarbasAl) [inline neon_core/gui/resting_screen.py:53]: better idea than comment above, we can skip the need to for a dummy MycroftSkill class if we use https://github.com/OpenVoiceOS/ovos_utils/blob/dev/ovos_utils/gui.py#L382

this is the same as the SkillGUI class but does not require a skill object, it is what i use in the ovos-common-play framework

in this case we init self.gui with only a skill_id string + the bus object

Comment (NeonDaniel) [reply neon_core/gui/resting_screen.py:53]: We would still need the event scheduling methods from MycroftSkill though (could be implemented separately, but I think scope of work is enough for a follow-up PR)

File: neon_core/gui/resting_screen.py (hunk 2 above)
Comment (NeonDaniel) [reply neon_core/gui/resting_screen.py:47]: Makes sense; good to reuse `MessageBusClient`s when we can

**Pull request merged**